src/Datadog.Trace/Configuration/StringConfigurationSource.cs

```diff
@@ -6,6 +9,39 @@ namespace Datadog.Trace.Configuration
     /// </summary>
     public abstract class StringConfigurationSource : IConfigurationSource
     {
+        /// <summary>
+        /// Returns a <see cref="IDictionary{TKey, TValue}"/> from parsing
+        /// <paramref name="data"/>.
+        /// </summary>
+        /// <param name="data">A string containing key-value pairs which are comma-separated, and for which the key and value are colon-separated.</param>
+        /// <returns><see cref="IDictionary{TKey, TValue}"/> of key value pairs.</returns>
+        public static IDictionary<string, string> ParseCustomKeyValues(string data)
+        {
+            var dictionary = new ConcurrentDictionary<string, string>();
+
+            if (string.IsNullOrWhiteSpace(data))
+            {
+                return dictionary;
```

Comment:
We should return `null` if `data` is `null` to be consistent with the other `Get` methods in this class. A null return value means the key was not present, and `CompositeConfigurationSource` depends on this behavior (it returns the first non-null value it finds). I think it's fine to return an empty dictionary if `data` is the empty string, though.

Reply:
👍